internal/generator/vector/output/syslog/syslog.go

```diff
@@ -345,3 +307,92 @@ var keyre = regexp.MustCompile(`^\$(\.[[:word:]]*)+$`)
 func IsKeyExpr(str string) bool {
 	return keyre.MatchString(str)
 }
+
+func buildDefaults(o *obs.Syslog) string {
+	var builder strings.Builder
+	type defaultRule struct {
+		cond     string
+		appName  string
+		procId   string
+		severity string
+		facility string
+	}
+	var defaultRules []defaultRule
+
+	if o.RFC == obs.SyslogRFC3164 {
+		if o.ProcId == "" || o.AppName == "" {
+			defaultRules = append(defaultRules, defaultRule{
+				cond:    `._internal.log_type == "infrastructure" && ._internal.log_source == "node"`,
+				appName: ifEmpty(o.AppName, nodeTag),
+				procId:  ifEmpty(o.ProcId, nodeProcId),
+			})
+		}
+		if o.AppName == "" || o.Severity == "" || o.Facility == "" {
+			defaultRules = append(defaultRules, defaultRule{
+				cond:     `._internal.log_source == "container"`,
+				appName:  ifEmpty(o.AppName, containerTag),
+				severity: ifEmpty(o.Severity, containerSeverity),
+				facility: ifEmpty(o.Facility, containerFacility),
+			}, defaultRule{
+				cond:     `._internal.log_type == "audit"`,
+				appName:  ifEmpty(o.AppName, auditTag),
+				severity: ifEmpty(o.Severity, auditSeverity),
+				facility: ifEmpty(o.Facility, auditFacility),
+			})
+		}
+	}
+
+	if o.RFC == obs.SyslogRFC5424 {
+		if o.MsgId == "" {
+			builder.WriteString(msgId + "\n")
+		}
+		if o.ProcId == "" || o.AppName == "" {
+			defaultRules = append(defaultRules, defaultRule{
+				cond:    `._internal.log_type == "infrastructure" && ._internal.log_source == "node"`,
+				appName: ifEmpty(o.AppName, nodeAppName),
+				procId:  ifEmpty(o.ProcId, nodeProcId2),
+			})
+		}
+		if o.AppName == "" || o.ProcId == "" || o.Severity == "" || o.Facility == "" {
```

Comment:
You can combine the above if statement with this one because they both check if `o.ProcId` or `o.AppName` are empty.